### PR TITLE
Mudit dev

### PR DESCRIPTION
**Pull request by mudit1990**
@doodhwala @rohinikapoor @sarankrishna920 

constants.py: added variables for user_emb_size, variable_emb_size and run_id

model1.py, model3.py: changed to use these constant variables

utils.py: save_embeddings() changed to incorporate run_id

w2v_glove.py: made it more generic for higher dimensions. Uses constants

### CODE REVIEW

File: code/constants.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
+import sys
 """
 This file is used to define constants used throughout the code
 """
 # ordering of the data columns in csv files
 SENDER_EMAIL = 0
 RECEIVER_EMAILS = 1
 EMAIL_BODY = 2
+
+# defining the dimensions for email, sender and receiver embedding
```
Comment (mohit-surana) [inline code/constants.py:10]: Really good idea to pull this out as a constant.
In the future, we might want to pass this as a command line argument as well.

File: code/main.py [hunk 2]
```diff
@@ -33,18 +33,17 @@
...
-    # model = Model1(epochs=10)
-    # model = Model2()
-    # model = Model3(epochs=50)
     # w2v = W2VCustom()
     w2v = W2VGlove()
 
-    # emails = dal.get_emails(NUM_EMAILS, fetch_all=True)
-    emails = dal.get_emails_by_users(num_users=num_users)
+    # emails = dal.get_emails(fetch_all=aTrue)
+    emails = dal.get_emails_by_users()
```
Comment (mohit-surana) [inline code/main.py]: Need to pass ```num_users``` parameter

File: code/model3.py [hunk 3]
```diff
@@ -25,14 +25,14 @@ def __init__(self, epochs=10):
         self.emailid_train_freq = {}
         super(Model3, self).__init__()
         # embedding lookup for 150 users each have 50 dimension representation
```
Comment (mohit-surana) [inline code/model3.py]: Stale doc string

File: code/w2v_glove.py [hunk 4]
```diff
@@ -3,13 +3,15 @@
 from gensim.scripts.glove2word2vec import glove2word2vec
 from gensim.models import KeyedVectors
 from gensim.models import Word2Vec
+import constants
 
 from w2v import W2V
 
+WORD_CORPUS_SIZE = '840B'
```
Comment (mohit-surana) [inline code/w2v_glove.py:10]: This should be pulled into constants.py
Perhaps programmatically get the corpus size corresponding to the dimension?

File: .gitignore [hunk 5]
```diff
@@ -6,6 +6,10 @@ __pycache__/
 # C extensions
 *.so
 
+# bin numpy files
```
Comment (mohit-surana) [inline .gitignore]: Need to remove the bin reference at the end of the gitignore.

File: code/utils.py [hunk 6]
```diff
@@ -156,7 +156,7 @@ def save_user_embeddings(email_ids, embeddings):
     the method saves the email embeddings passed to it as a pickle file
     :return: saves as pickle file
     """
-    filepath = '../resources/embeddings.pkl'
+    filepath = '../resources/' + 'embeddings_' + constants.RUN_ID + '.pkl'
```
Comment (mohit-surana) [inline code/utils.py:159]: Now we will have a lot of embeddings corresponding to each experiment
Should we save them in the log folder?

**Comment by mudit1990** [comment]
made the review comment changes. changing model2 to use constants file ... i'll do in another commit

**Pull request merged**